packages/framer-motion-3d/src/render/use-render.ts

```diff
@@ -43,12 +43,14 @@ function useVisualProps(props: ThreeMotionProps & MeshProps) {
         const visualProps: ThreeMotionProps & MeshProps = {}
 
         for (const key in props) {
-            const prop = props[key]
+            const prop = props[key as keyof typeof props]
```

Comment:
@mattgperry how about depending on:
- return type
- and global per-repo type override for loops, so the key is auto-inferred from the array
  - you should also consider to switch to Object.keys for a slight performance boost 
  - https://dev.to/tmhao2005/typescript-override-typing-for-objectkeys-5dmh

It would ensure cases like this don’t need manual `as` override, removing possibilities of human errors.

If you still want to manually define `key`’s type, then you should do it [once before the for loop](https://stackoverflow.com/a/59656330).